### PR TITLE
Add WordCheck unit tests

### PR DESCRIPTION
**Pull request by cpeel**
As part of updating WordCheck to support UTF-8 I realized that there are no unit tests for WordCheck. I stepped back to add some and realized that some of the WordCheck functions, notably `get_bad_words_for_text()` was not unit-testable.

This series of commits reorganizes some of the WordCheck functions such that they are testable and then ultimately actually tests them.

### CODE REVIEW

File: pinc/wordcheck_engine.inc [hunk 1]
```diff
@@ -1020,6 +1018,31 @@ function generate_frequencies($wordList)
     return $wordCount;
 }
 
+// Given an array of langauge names, return a list of valid langcode3s.
+// Arguments:
+//      langauges - array of langauge names
```
Comment (mlazaric) [inline pinc/wordcheck_engine.inc]: Small typo "langauges".

File: pinc/wordcheck_engine.inc [hunk 2]
```diff
@@ -338,15 +354,17 @@ function get_bad_words_via_external_checker($input_words_w_freq, $projectid, $im
 // returns a list of 'bad' words in a text based on some pattern
 // Arguments:
 //   input_words_w_freq - an array whose keys are the distinct words of the input text
-//   langcode3s - codes for languages to check against
+//   languages - array of langauges to check against
```
Comment (mlazaric) [inline pinc/wordcheck_engine.inc]: Small typo "langauges".

File: pinc/wordcheck_engine.inc (hunk 1 above)
Comment (cpeel) [reply pinc/wordcheck_engine.inc]: Good catch, thanks!

File: pinc/wordcheck_engine.inc (hunk 2 above)
Comment (cpeel) [reply pinc/wordcheck_engine.inc]: Thanks!

**Comment by cpeel** [comment]
I just addressed (yet another) pair of typos. Unless someone thinks differently I consider this ready for merging.

**Pull request merged**